src/main/java/ca/nrc/cadc/beacon/web/resources/MainPageServerResource.java

```diff
@@ -171,6 +171,19 @@ Representation representFolderItem(final FolderItem folderItem, final Iterator<S
             dataModel.put("startURI", startNextPageURI.toString());
         }
 
+        // Add the current VOSpace service name so that navigation links can be rendered correctly
+        String vospaceSvcName = getCurrentVOSpaceService();
+        if (StringUtil.hasLength(vospaceSvcName)) {
+            dataModel.put("vospaceSvcPath", vospaceSvcName + "/");
+            dataModel.put("vospaceSvcName", vospaceSvcName);
+        } else {
+            dataModel.put("vospaceSvcPath", "/");
+            dataModel.put("vospaceSvcName", "");
```

Comment:
Won't there always be a default VOSpace service?

Reply:
Yes, there will be. Might have overdone it, considering the old URLs (/storage/list, etc.) need to be supported as input to the service. And you're right, they don't need to be printed to the screen through the templates anymore.

Reply:
Added a check at the point app config is read in for a default service name - that was the only hole in the plan that a default is required.